src/blocks/list.js

```diff
@@ -218,5 +218,51 @@ module.exports = Block.extend({
       return `<li>${item.content}</li>`;
     }).join("\n");
     return `<ul>${listItems}</ul>`;
+  },
+
+  splitCurrentEditor: function(scribe) {
+    if (scribe.getTextContent().length === 0) return false;
+
+    var content = rangeToHTML(selectToEnd(scribe));
+    this.addListItemAfterCurrent(content);
+    return true;
+  },
+
+  split: function() {
+    const currentEditor = this.getCurrentTextEditor();
+    if (!currentEditor) return;
+
+    var scribe = currentEditor.scribe;
+    if (this.splitCurrentEditor(scribe)) {
+      this.focusOnNeighbor();
+      this.addListItemAfterCurrent("");
+    }
+    this.splitListItem(this.getCurrentTextEditor().scribe);
+  },
+
+  splitListItem: function(scribe, options = {}) {
```

Comment:
Like you said before, starting to remember all the complexity that started to come out when we were working on Sir Trevor 2.

Reply:
Yes. Each block with text that can be passed into will need to know how to split. Otherwise the content will be placed after that block.